components/director/internal/domain/viewer/converter.go

```diff
@@ -0,0 +1,23 @@
+package viewer
+
+import (
+	"github.com/kyma-incubator/compass/components/director/internal/consumer"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/pkg/errors"
+)
+
+func ToViewer(cons consumer.Consumer) (*graphql.Viewer, error) {
```

Comment:
The first one , `SystemAuthReferenceObjectType` is used in SystemAuth logic. Additionally we cannot add `USER`, because it will introduce confusion.

`Consumer` type is used in TenantMappingService and it describe who is using our API. I would call it `input` type. 

`Viewer` type is graphql type and it's only used in Viewer. I treat it as `response` type.

From my perspective I would keep it as it's because those types are used in different contexts. Additionally those different types, describe precisely how they should be used and what they represent.
I would look at those types, like `application` entity. We have `ApplicationRegisterInput` and `Application`.
I am aware that this is duplication of code.